icicle-compiler/src/Icicle/Runtime/Data/Array.hs

```diff
@@ -297,9 +297,24 @@ grow pool arrayOld@(Array ptrOld) lengthNew = do
     !capacityNew =
       calculateCapacity lengthNew
 
-  if capacityOld >= capacityNew then do
+  if lengthOld > lengthNew then do
+    let
+      deadSize =
+        (lengthOld - lengthNew) * elementSize
+
+      deadPtr =
+        unsafeElementPtr arrayOld `plusPtr`
+        fromIntegral (lengthNew * elementSize)
+
+    c_memset deadPtr 0 (fromIntegral deadSize)
```

Comment:
that's really cool! does that mean grow should zero out everything between the old count and the new count?